include/vcpkg/installedpaths.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <vcpkg/base/files.h>
+
+#include <vcpkg/binaryparagraph.h>
+#include <vcpkg/packagespec.h>
+#include <vcpkg/triplet.h>
+
+namespace vcpkg
+{
+    struct InstalledPaths
+    {
+        explicit InstalledPaths(Path&& root) : m_root(std::move(root)) { }
+
+        const Path& root() const { return m_root; }
+        Path listfile_path(const BinaryParagraph& pgh) const;
+        const Path& installed() const;
+
+        Path vcpkg_dir() const { return m_root / "vcpkg"; }
+        Path vcpkg_dir_status_file() const { return vcpkg_dir() / "status"; }
+        Path vcpkg_dir_info() const { return vcpkg_dir() / "info"; }
+        Path vcpkg_dir_updates() const { return vcpkg_dir() / "updates"; }
+        Path lockfile_path() const { return vcpkg_dir() / "vcpkg-lock.json"; }
```

Comment:
Depending on how often these are used it may be a good idea to precalculate them and return const&s rather than returning prvalues. (No change requested)